### PR TITLE
docs(effects): add guide for functional effects

### LINKED ISSUE #3798: Add docs for functional effects
### Information

Add docs for functional effects that are introduced in v15.2.

### Documentation page

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by markostanimirovic**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #3798 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/effects/index.md [hunk 1]
```diff
@@ -172,17 +178,67 @@ export class MovieEffects {
...
+    );
+  },
+  { functional: true, dispatch: false }
+);
+</code-example>
 
-After you've written your Effects class, you must register it so the effects start running. To register root-level effects, add the `EffectsModule.forRoot()` method with an array of your effects to your `AppModule`.
+<div class="alert is-important">
+
+It's recommended to inject all dependencies as effect function arguments for easier testing. However, it's also possible to inject dependencies in the effect function body. In that case, the `inject` function must be called within the synchronous context.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/effects/index.md:225]: Should we point to the Angular docs here?

```suggestion
It's recommended to inject all dependencies as effect function arguments for easier testing. However, it's also possible to inject dependencies in the effect function body. In that case, the [`inject` function](https://angular.io/api/core/Inject) must be called within the synchronous context.
```

File: projects/ngrx.io/content/guide/effects/testing.md [hunk 2]
```diff
@@ -335,3 +335,37 @@ it('should get customers', () => {
...
+      ActorsApiActions.actorsLoadedSuccess({ actors: actorsMock })
+    );
+    done();
+  });
+});
+</code-example>
+
+<div class="alert is-helpful">
+
+Check the `loadActors` effect implementation [here](guide/effects#functional-effects).
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/effects/testing.md:369]:
```suggestion
You can check the `loadActors` effect implementation [here](guide/effects#functional-effects).
```

File: projects/ngrx.io/content/guide/effects/index.md [hunk 3]
```diff
@@ -172,17 +178,67 @@ export class MovieEffects {
...
+import { ActorsService } from './actors.service';
+import { ActorsPageActions } from './actors-page.actions';
+import { ActorsApiActions } from './actors-api.actions';
+
+export const loadActors = createEffect(
+  (actions$ = inject(Actions), actorsService = inject(ActorsService)) => {
+    return actions$.pipe(
+      ofType(ActorsPageActions.opened),
+      exhaustMap(() =>
+        actorsService.getAll().pipe(
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/effects/index.md:200]: Should this import the RxJS operators, or would this make it bloated?

**Review by timdeschryver — APPROVED** [review]
LGTM, but I left a few nits.

**Review by brandonroberts — APPROVED** [review]
LGTM

**Pull request merged**